### PR TITLE
Adding documentation for c++

### PR DESCRIPTION
**Pull request by senior7515**

### CODE REVIEW

File: source/quick_start_cpp.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   Debug mode
+   scons: done reading SConscript files.
+   scons: Building targets ...
+   g++-4.9 -o word_counter.o -c -g3 -fno-omit-frame-pointer -fvar-tracking-assignments -gdwarf -O1 -std=c++1y -Wall -Wnon-virtual-dtor -Werror -Wextra -Wformat -pedantic -Wmissing-braces -Wparentheses -ftemplate-depth-256 -Woverloaded-virtual -Wpointer-arith -Wformat-security -Wunused -Wno-unused-parameter -Wcast-align -Wmissing-field-initializers -DFOLLY_HAVE_PTHREAD_ATFORK -DDEBUG word_counter.cc
+   CONCORD...... linking... word_counter
+   g++-4.9 -o word_source.o -c -g3 -fno-omit-frame-pointer -fvar-tracking-assignments -gdwarf -O1 -std=c++1y -Wall -Wnon-virtual-dtor -Werror -Wextra -Wformat -pedantic -Wmissing-braces -Wparentheses -ftemplate-depth-256 -Woverloaded-virtual -Wpointer-arith -Wformat-security -Wunused -Wno-unused-parameter -Wcast-align -Wmissing-field-initializers -DFOLLY_HAVE_PTHREAD_ATFORK -DDEBUG word_source.cc
+   CONCORD...... linking... word_source
+   scons: done building targets.
+
+You should see a directoy layout like this:
```
Comment (RBlafford) [inline source/quick_start_cpp.rst]: directory

File: source/quick_start_cpp.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+Downloading our Getting Started Package
+---------------------------------------
+
+.. code-block:: bash
+
+   $ git clone https://github.com/concord/getting-started.git concord-getting-started
+   $ cd concord-getting-started
+
+Start the containerand the cluster with the provided scripts
```
Comment (bigs) [inline source/quick_start_cpp.rst]: container and

Comment (senior7515) [inline source/quick_start_cpp.rst]: done

File: source/quick_start_cpp.rst [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+     2 gems installed
+     Launching Master
+     Launching Slave
+     Launching Scheduler
+     [addJava] arg = '-DMESOS_NATIVE_JAVA_LIBRARY=/usr/local/lib/libmesos.so'
+     [addJava] arg = '-Djava.library.path=/usr/local/lib'
+     Launching Dashboard
+     ...
+
+.. note:: This image is intented for testing our client API. Do not run a
```
Comment (bigs) [inline source/quick_start_cpp.rst]: does this compile? pretty sure you need two newlines after a directive i.e.

```rst
.. note::

  My note here
```

File: source/quick_start_cpp.rst (hunk 1 above)
Comment (senior7515) [inline source/quick_start_cpp.rst]: done.

File: source/quick_start_cpp.rst (hunk 3 above)
Comment (senior7515) [inline source/quick_start_cpp.rst]: ya. this works.

File: source/quick_start_cpp.rst [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+Check the Mesos Dashboard
+-------------------------
+Go to http://localhost:5050 in your browser and you will see the two
+computations running on Concord:
+
+.. image:: images/mesos-with-computations.png
+
+You should be able to find the logs by clicking into the **Sandbox** for each
+task you've started!
```
Comment (bigs) [inline source/quick_start_cpp.rst]: maybe elaborate that we log to stderr

**Pull request merged**